crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -57,6 +59,56 @@ fn channel_balance_storage_key(channel: &ChannelId, asset: asset::Id) -> String
 
 #[async_trait]
 pub(crate) trait StateReadExt: StateRead {
+    #[instrument(skip(self))]
+    async fn get_account_balances(&self, address: Address) -> Result<Vec<AssetBalance>> {
+        use crate::asset::state_ext::StateReadExt as _;
+
+        let prefix = format!("{}/balance/", storage_key(&address.encode_hex::<String>()));
+        let mut balances: Vec<AssetBalance> = Vec::new();
+
+        let mut stream = std::pin::pin!(self.prefix_keys(&prefix));
+        while let Some(Ok(key)) = stream.next().await {
```

Comment:
Q about the entire body of the loop: should this entire block fail if some account balances could not be retrieved? Or is it OK to log these and carry on?

Reply:
I think it's fine as this is just used in the CLI right now